### PR TITLE
Add agent group model

### PR DESCRIPTION
**Pull request by elyousfi5**

### CODE REVIEW

**Review by benyissa — APPROVED** [review]
LGTM

File: tests/runtimes/local/models/models_test.py [hunk 1]
```diff
@@ -158,3 +158,73 @@ def testModelsVulnerability_whenRiskRatingIsCritcal_doNotRaiseError(
...
+    """Test Agent save implementation."""
+    mocker.patch.object(models, "ENGINE_URL", db_engine_path)
+    models.Agent.create("test")
+
+    with models.Database() as session:
+        assert session.query(models.Agent).count() == 1
+        assert session.query(models.Agent).all()[0].key == "test"
+
+
+def testModelsAgentArgument_always_createsAgentArgument(mocker, db_engine_path):
```
Comment (PiranhaSa) [inline tests/runtimes/local/models/models_test.py:173]: type please

File: src/ostorlab/runtimes/local/models/models.py [hunk 2]
```diff
@@ -127,6 +127,9 @@ class Scan(Base):
     asset = sqlalchemy.Column(sqlalchemy.String(255))
     created_time = sqlalchemy.Column(sqlalchemy.DateTime)
     progress = sqlalchemy.Column(sqlalchemy.Enum(ScanProgress))
+    agent_group_id = sqlalchemy.Column(
+        sqlalchemy.Integer, sqlalchemy.ForeignKey("agent_group.id")
+    )
```
Comment (mohsinenar) [inline src/ostorlab/runtimes/local/models/models.py:132]: Is this nullable, waht will happen for the existing scans?

File: tests/runtimes/local/models/models_test.py (hunk 1 above)
Comment (elyousfi5) [reply tests/runtimes/local/models/models_test.py:173]: applied

File: src/ostorlab/runtimes/local/models/models.py (hunk 2 above)
Comment (elyousfi5) [reply src/ostorlab/runtimes/local/models/models.py:132]: Iti is nullable
image

File: src/ostorlab/runtimes/local/models/alembic/env.py [hunk 3]
```diff
@@ -73,7 +73,9 @@ def run_migrations_online() -> None:
     )
 
     with connectable.connect() as connection:
-        context.configure(connection=connection, target_metadata=target_metadata)
+        context.configure(
+            connection=connection, target_metadata=target_metadata, render_as_batch=True
```
Comment (3asm) [inline src/ostorlab/runtimes/local/models/alembic/env.py:77]: Why the change?

**Review by 3asm — APPROVED** [review]
